### PR TITLE
Add first html version of the build verdict

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

File: subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/writer/BuildVerdictConfigurationHtml.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import kotlinx.html.body
+import kotlinx.html.h2
+import kotlinx.html.h3
+import kotlinx.html.head
+import kotlinx.html.html
+import kotlinx.html.pre
+import kotlinx.html.stream.createHTML
+import kotlinx.html.title
+
+internal fun BuildVerdict.Configuration.html(): String {
```
Comment (sergioboi) [inline subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/writer/BuildVerdictConfigurationHtml.kt:14]: I will add screenshots of the first version

File: subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/BuildFailedListener.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.avito.android.build_verdict.internal
+
+import com.avito.android.build_verdict.internal.writer.BuildVerdictWriter
+
+internal class BuildFailedListener(
```
Comment (dsvoronin) [inline subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/BuildFailedListener.kt:5]: why do you think this abstraction better than just provide list of writers to every failure listener?

File: subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/writer/PlainTextBuildVerdictWriter.kt [hunk 3]
```diff
@@ -2,29 +2,23 @@ package com.avito.android.build_verdict.internal.writer
...
                 is BuildVerdict.Execution -> buildVerdict.plainText()
                 is BuildVerdict.Configuration -> buildVerdict.plainText()
             }
         )
-        logger.warn("Pretty build verdict at $file")
     }
 
     companion object {
-        const val buildVerdictFileName = "pretty-build-verdict.txt"
+        const val fileName = "pretty-build-verdict.txt"
```
Comment (dsvoronin) [inline subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/writer/PlainTextBuildVerdictWriter.kt:22]: it looks like a constructor property for writers, why static?

File: subprojects/gradle/build-verdict/src/test/kotlin/com/avito/android/build_verdict/BaseBuildVerdictTest.kt [hunk 4]
```diff
@@ -14,8 +14,9 @@ import com.google.gson.JsonElement
 import org.junit.jupiter.api.io.TempDir
 import java.io.File
 import java.lang.reflect.Type
-import com.avito.android.build_verdict.internal.writer.PlainTextBuildVerdictWriter.Companion.buildVerdictFileName as plainTextVerdict
-import com.avito.android.build_verdict.internal.writer.RawBuildVerdictWriter.Companion.buildVerdictFileName as rawVerdict
+import com.avito.android.build_verdict.internal.writer.HtmlBuildVerdictWriter.Companion.fileName as htmlFileName
```
Comment (dsvoronin) [inline subprojects/gradle/build-verdict/src/test/kotlin/com/avito/android/build_verdict/BaseBuildVerdictTest.kt]: inconsistent naming, all 3 should end with `Verdict` or `FileName`

**Comment by dsvoronin** [comment]
Some essential tests for HtmlVerdict will be nice to have

File: subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/BuildFailedListener.kt (hunk 2 above)
Comment (sergioboi) [reply subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/BuildFailedListener.kt:5]: Right now we just dump the verdict to a file, but I guess in the future we might decide to write it to the logs, etc. So I want a verdict callback and decide what I will do there.

File: subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/writer/PlainTextBuildVerdictWriter.kt (hunk 3 above)
Comment (sergioboi) [reply subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/internal/writer/PlainTextBuildVerdictWriter.kt:22]: It's used in tests and as a constructor argument

**Comment by sergioboi** [comment]
Examples of current version
Screenshot 2020-12-17 at 15 25 13
Screenshot 2020-12-17 at 15 25 23
Screenshot 2020-12-17 at 15 25 31
Screenshot 2020-12-17 at 15 25 39
Screenshot 2020-12-17 at 15 25 47
Screenshot 2020-12-17 at 15 26 07
Screenshot 2020-12-17 at 15 26 14

**Comment by eugene-krivobokov** [comment]
> No error logs

This message is highlighted but looks normal.

**Review by eugene-krivobokov — APPROVED** [review]

**Pull request merged**